app/controllers/admin/ministries_controller.rb

```diff
@@ -36,14 +59,35 @@ def new
 	end
 
 	def create
-    @ministry = Ministry.new(secure_params.merge(created_by_id: current_user.id))
-    if @ministry.save
-      redirect_to admin_ministry_path(@ministry), flash_success_info: "Ministry was successfully created."
-    else
-    	flash[:flash_info] = "Ministry was not successfully created."
-      render :new
-    end
-  end
+		@ministry = Ministry.new(secure_params.merge(created_by_id: current_user.id))
+		
+		if @ministry.location_id.nil?
+			@ministry.location_id = 0
+		else
+			@location_type = Location.find(@ministry.location_id).location_type
+			if @ministry.level == "national"
```

Comment:
Instead of doing this, add a custom validation error on the model.

Reply:
We are not sure on how to add custom validations since we are not experts on Ruby. Please take care of it at your end.

Reply:
Did go through the documentation on how to add custom validations but have not been able to make any changes as we have come to an end on the FFG program. Please take care of it at your end.